A33-Fault-Injection.md

```diff
@@ -0,0 +1,262 @@
+Client-Side Fault Injection
+----
+* Author(s): lidiz
+* Approver: roth, ejona, dfawley
+* Status: In Review
+* Implemented in: all languages
+* Last updated: August 13th, 2020
+* Discussion at: https://groups.google.com/g/grpc-io/c/d9dI5Hy9zzk
+
+## Abstract
+
+Fault injection is an essential feature that assists developers in building fault-tolerant applications. To keep gRPC’s competitive advantage, this doc aims to design a fault injection mechanism on the client side for gRPC while allowing transparent transition from Envoy configuration.
+
+## Background
+
+### Envoy HTTPFilter
+
+Envoy has a network level filter named HTTP Connection Manager. It is responsible to handle common functionalities against HTTP connections and requests. In its `http_filters` field, users can set up multiple HTTPFilters. Each HTTPFilter requires at least a name. The name will be the identifier that assists future updates of filter configuration. For example, users can specify an `{"name": "envoy.cors"}` filter in `http_filters` with no content, then update the CORS filter’s configuration with the `typed_per_filter_config` field of a VirtualHost in RDS.
```

Comment:
Do we need this or can we just reference an Envoy page that describes the HTTP CM and HTTP filter?

Reply:
Thanks for the suggestion. I have removed this paragraph, and added two links instead.